src/components/User.vue

```diff
@@ -1,38 +1,144 @@
-<script setup lang="ts">
-  import { createApp } from 'vue';
+<script lang="ts">
+import { defineComponent, ref, watchEffect } from 'vue';
+import Search from './Search.vue';
 
+interface UserInfo {
+  avatar_url: string;
+  name: string | null;
+  login: string;
+  created_at: string;
+  bio: string | null;
+  public_repos: number;
+  followers: number;
+  following: number;
+  location: string | null;
+  blog: string | null;
+  twitter_username: string | null;
+  company: string | null;
+}
+
+export default defineComponent({
+  components: {
+    Search,
+  },
+  setup() {
+    const username = ref('octocat');
+    const userInfo = ref<UserInfo | null>(null);
+    const error = ref<any>(null);
+
+    async function fetchUserInfo(username: string) {
+      try {
+        const response = await fetch(`https://api.github.com/users/${username}`);
+        if (!response.ok) {
+          const errorData = await response.json();
+          error.value = `Error fetching user information: ${response.status} - ${errorData.message}`;
```

Comment:
In this case, it would be better, that user would see something more like "User not found" when the response is 404. And a more detailed error - if it throws an exception.